queue/QueueConsumer.js

```diff
@@ -43,12 +115,30 @@ module.exports = class QueueConsumer extends BaseQueue {
     // messages that are being handled at once)
     await channel.prefetch(this._prefetchCount)
 
-    // TODO: other assertions may need ot be done on the queue
-    await channel.assertQueue(queueName, {
+    // create exchange for dead letter exchange,
+    // we specify the "fanout" so that messages published to
+    // this exchange sent to all queues that are bound
+    // to the exchange.
+    //
+    // In this case, we only have a single queue to listening
+    // for this exchange
+    await channel.assertExchange(deadLetterExchangeName, AMQ_FANOUT)
+
+    // assert the main queue to consume from
+    await assertQueue(channel, queueName)
+
+    // create dead letter queue to store messages that will
+    // be analyzed later
+    //
+    // (use regular assert queue here since this is unique to consumer)
+    await channel.assertQueue(deadLetterQueueName, {
       durable: true,
       noAck: false
     })
 
+    // bind the queue and the dead letter exchange together
+    await channel.bindQueue(queueName, deadLetterExchangeName, 'key')
```

Comment:
Cool concept. You can have a channel bind a queue to an exchange. This will make it so that you can listen to messages that are sent to both a queue and an exchange. Exchanges allow for different ways to handle messages.

For ex. you can make an exchange and bind it to multiple queues. Then, depending on your exchange type, you can set up different ways for messages to get sent to those queues.  So if you want direct routing, you can set up the exchange to send to certain queues via a routing key. Or alternatively you can set an exchange up publish to multiple queues. Seems like it could be useful in the future.